packages/tuono/src/shared/RouterContextProviderWrapper.tsx

```diff
@@ -0,0 +1,32 @@
+import type { JSX } from 'react'
+import { RouterProvider } from 'tuono-router'
+import type { RouterInstanceType } from 'tuono-router'
+
+import { useTuonoContextServerPayload } from './TuonoContext'
+
+interface RouterContextProviderWrapperProps {
+  router: RouterInstanceType
+}
+
+/**
+ * This component is needed to get the data from {@link TuonoContext}
+ * since the provider is also located in {@link TuonoEntryPoint}
+ * hence the context cannot be accessed directly there
+ *
+ * @see https://github.com/tuono-labs/tuono/issues/410
+ */
+export function RouterContextProviderWrapper(
+  props: RouterContextProviderWrapperProps,
```

Comment:
Why not destructuring it directly in the argument?
```suggestion
  { router }: RouterContextProviderWrapperProps,
```